presto-docs/src/main/sphinx/admin/resource-groups.rst

```diff
@@ -67,6 +67,12 @@ Resource Group Properties
 * ``jmxExport`` (optional): If true, group statistics are exported to JMX for monitoring.
   Defaults to ``false``.
 
+* ``queuedTimeLimit`` (optional): maximum amount of time a query can get queued in this group before
+  it is marked as failed.
+
+* ``runningTimeLimit`` (optional): maximum amount of time a query can run in this group before
+  it is marked as failed.
```

Comment:
Does it need to be clarified that it excludes queue time?

Reply:
The *"can run in this group"* part is confusing since it's different than the queued limit which is cumulative across multiple groups. How about

*"maximum amount of time a query in this group can execute (not including queued time). This value only applies to the matching resource group for the query; it is not inherited or combined from ancestor groups."*

Note that I don't actually know if this is how it works.

Reply:
I am not the author of this feature, but looking at the [code](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java#L798):

- I am not sure what you mean by "queued limit which is cumulative across multiple groups", but nothing is accumulated across groups. According to the source if a query's elapsed time exceeds the `queuedTimeLimit` of the resource group it's queued in then it is marked as failed.

- I think `it is not inherited or combined from ancestor groups.` is also confusing, because ancestor groups of a leaf group doesn't execute/queue queries, it's only the leaf groups that can queue/execute queries, and it's the leaf groups that will enforce these limits.

I will clarify to exclude the queueing time.

Reply:
I think it is strange to say that resource groups executes queries. Per the introductory paragraph, "A query belongs to a single resource group". We should stick with the same terminology.

How about changing *"a query can execute in this group"* to *"a query in this group can execute"*?